contracts/tokens/SecurityToken.sol

```diff
@@ -721,4 +741,50 @@ contract SecurityToken is StandardToken, DetailedERC20, ReentrancyGuard, Registr
         return _getValueAt(checkpointBalances[_investor], _checkpointId, balanceOf(_investor));
     }
 
+    /**
+     * @notice Use by the issuer ot set the controller addresses
+     * @param _controller address of the controller
+     */
+    function setController(address _controller) public onlyOwner {
+        require(!controllerFrozen);
+        emit LogSetController(controller, _controller);
+        controller = _controller;
+    }
+
+    /**
+     * @notice Permanently freeze controller functionality of this security token.
+     */
+    function freezeController() external isEnabled("freezeControllerAllowed") onlyOwner {
```

Comment:
This can be simplified. We don't need to freeze a controller. The setController function could be used to make the controller 0x0 so there's no longer a controller if you want to remove him/her.

Reply:
This logic is needed to notify that the controller will never be used again

Reply:
why?